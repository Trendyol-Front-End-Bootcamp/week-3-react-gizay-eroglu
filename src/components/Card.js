import { useHistory } from "react-router-dom";

function Card({ id, name, status, species, gender, image }) {
  const history = useHistory();

  const handleOnClickCard = (event) => {
    history.push(`character/${id}`);
  };

  return (
    <div className='card' onClick={handleOnClickCard}>
      <img src={image} alt='Character' />
      <div className='card-text'>
        <h5>{name}</h5>
        <small>{status}</small>
        <small>{species}</small>
        <small>{gender}</small>
      </div>
    </div>
  );
}

export default Card;
