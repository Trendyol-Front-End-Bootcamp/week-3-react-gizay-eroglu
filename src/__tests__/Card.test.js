import React from "react";
import Adapter from "enzyme-adapter-react-16.1";
import { shallow, configure } from "enzyme";

import Card from "../components/Card";

configure({adapter: new Adapter()});

describe("Card component test", function () {

  let wrapper;

  beforeEach(() => { 
    wrapper = shallow(<Card />);
  })

  it("should render card", () => {
    expect(wrapper).toMatchSnapshot();
  })
  
  it("should check class of div element which includes card texts", () => {
    const classOfDiv = wrapper.find("div div").hasClass("card-text");
    expect(classOfDiv).toEqual(true);
  })

  it("should give response to click event", () => {
     const mockCallBackClick = jest.fn();
        const wrapper = shallow(<div className='card' onClick={mockCallBackClick}></div>);
        wrapper.find('div').simulate('click');
        expect(mockCallBackClick.mock.calls.length).toEqual(1);
  });

});