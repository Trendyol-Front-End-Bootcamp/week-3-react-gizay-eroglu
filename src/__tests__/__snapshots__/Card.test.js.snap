// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component test should render card 1`] = `ShallowWrapper {}`;
